src/coap_block.c

```diff
@@ -3244,13 +3244,25 @@ coap_handle_request_put_block(coap_context_t *context,
     if (lg_srcv->total_len < saved_offset + length) {
       lg_srcv->total_len = saved_offset + length;
     }
-    lg_srcv->body_data = coap_block_build_body(lg_srcv->body_data, length, data,
-                                               saved_offset, lg_srcv->total_len);
-    if (!lg_srcv->body_data) {
-      coap_add_data(response, sizeof("Memory issue")-1,
-                    (const uint8_t *)"Memory issue");
-      response->code = COAP_RESPONSE_CODE(500);
-      goto skip_app_handler;
+    if (context && context->block_data_handler && !resource->is_proxy_uri &&
+        !resource->is_reverse_proxy) {
+      coap_pdu_code_t resp_code;
+      resp_code = context->block_data_handler(session, pdu, resource, &lg_srcv->body_data,
+                                              length, data, saved_offset,
+                                              lg_srcv->total_len);
+      if (resp_code != COAP_EMPTY_CODE) {
+        response->code = resp_code;
+        goto skip_app_handler;
```

Comment:
Using `coap_response_t`, I would send back the `COAP_RESPONSE_CODE(500);` here, rather than doing it in the application.